ec2_vol: Add key selection support for EBS volume encryption.

bachp opened the pull request:
Original-code-by: https://github.com/nonshankus
More or less a rebase of PR onto master: https://github.com/ansible/ansible-modules-core/issues/3217

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ec2_vol

##### ANSIBLE VERSION
```
ansible 2.1.1.0
```

##### SUMMARY
Add support to specify the encryption key for encrypted EBS volumes.

It adds an additional config `kms_key_id` that can be used to use a key other than the default.

nonshankus committed:
Fixes: #3217 - Add a new parameter for the kms_key_id to the ec2_vol module.

nonshankus committed:
Fixes: #3217 - Correcting comment string as requested.

nonshankus committed:
Fixes: #3217 - Adding boto version when kms_key_id is used. Also re-adding accidentally removed comment line.

ryansb commented on lib/ansible/modules/cloud/amazon/ec2_vol.py:
```diff
@@ -546,6 +565,9 @@ def main():
     if encrypted and not boto_supports_volume_encryption():
         module.fail_json(msg="You must use boto >= v2.29.0 to use encrypted volumes")
 
+    if kms_key_id and not boto_supports_kms_key_id():
```
This check should be `kms_key_id is not None`

ryansb commented on lib/ansible/modules/cloud/amazon/ec2_vol.py:
```diff
@@ -335,7 +349,10 @@ def create_volume(module, ec2, zone):
         try:
             if boto_supports_volume_encryption():
-                volume = ec2.create_volume(volume_size, zone, snapshot, volume_type, iops, encrypted)
+                if kms_key_id:
```
This check should be `kms_key_id is not None`

willthames commented:
:-1: I see this as misappropriation of a commit.

It's not hard to rebase a commit while retaining original authorship. If further rework is required on top, you can add your own commit to fix things.

bachp commented:
@willthames Could you point me to some instructions on how to do the rebase? The problem I was facing is that the location of the files changed when they were moved to the ansible repository.

willthames commented:
@bachp sure, no problem, here we go:

```
cd path/to/ansible-modules-core
git fetch upstream pull/3218/head:pr-3218
git checkout pr-3218
git format-patch upstream/devel -o /tmp
```

Then to apply that same patch

```
cd path/to/ansible
git checkout -b ec2_enc_keys devel
git am -p1 --directory lib/ansible/modules /tmp/000*
```

bachp committed:
Cleanup of EBS volume key pull request

bachp commented:
@ryansb I addressed your issues.

@willthames Thanks the `--directory` option was the one I was not aware of. Will sure be helpful in the future again.

willthames commented:
ready_for_review

nbasinge commented:
What is the status? Thanks.

ryansb commented:
LGTM, just needs a maintainer approval.

willthames commented:
shipit

willthames commented:
@ryansb not sure community review has been requested here - ansibot hasn't cc'd anyone for review.

willthames commented:
!needs_revision
